### DISCUSSION TITLE
`@property` and types

### DISCUSSION THREAD

**Discussion by patrick91** [microsoft/pyright#7535] [General]
Hi there! I was trying to use the new `default` param in `TypeVar`, but my use cases uses the type var on a property, like this:

```python
from typing_extensions import TypeVar

T = TypeVar("T", default=int)


class C[T]:
    @property
    def prop(self) -> T:
        return ...


reveal_type(C[int].prop)
```

`reveal_type` says that .prop is a `property`, which is correct, but I'd love to be able to check the actual type 😊

Is there a way to do so?

**Comment by Viicos** [comment]
```python
reveal_type(C[int]().prop)
```

should probably work. `C[int].prop` refers to the actual `property` object as `prop` is accessed from the class object instead of an instance.

**Reply by patrick91** [reply]
ah, I'm silly! thanks!
